tests/helper_test.py

```diff
@@ -0,0 +1,233 @@
+# encoding=utf-8
+# Copyright © 2016-2017 Dylan Baker
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test suite for alot.helper module."""
+
+from __future__ import absolute_import
+
+import datetime
+import random
+import unittest
+
+import mock
+
+from alot import helper
+
+
+class Test_split_commandline(unittest.TestCase):
+
+    def _test(self, base, expected):
+        """Shared helper to reduce some boilerplate."""
+        actual = helper.split_commandline(base)
+        self.assertListEqual(actual, expected)
+
+    def test_simple(self):
+        base = 'echo "foo";sleep 1'
+        expected = ['echo "foo"', 'sleep 1']
+        self._test(base, expected)
+
+    def test_single(self):
+        base = 'echo "foo bar"'
+        expected = [base]
+        self._test(base, expected)
+
+    def test_unicode(self):
+        base = u'echo "foo";sleep 1'
+        expected = ['echo "foo"', 'sleep 1']
+        self._test(base, expected)
+
+
+class Test_split_commandstring(unittest.TestCase):
+
+    def _test(self, base, expected):
+        """Shared helper to reduce some boilerplate."""
+        actual = helper.split_commandstring(base)
+        self.assertListEqual(actual, expected)
+
+    def test_bytes(self):
+        base = b'echo "foo bar"'
+        expected = [b'echo', b'foo bar']
+        self._test(base, expected)
+
+    def test_unicode(self):
+        base = u'echo "foo €"'
+        expected = [b'echo', u'foo €'.encode('utf-8')]
+        self._test(base, expected)
+
+
+class Test_string_sanitize(unittest.TestCase):
+
+    def test_tabs(self):
+        base = 'foo\tbar\noink\n'
+        expected = 'foo' + ' ' * 5 + 'bar\noink\n'
+        actual = helper.string_sanitize(base)
+        self.assertEqual(actual, expected)
+
+
+class Test_string_decode(unittest.TestCase):
+
+    def _test(self, base, expected, encoding='ascii'):
+        actual = helper.string_decode(base, encoding)
+        self.assertEqual(actual, expected)
+
+    def test_ascii_bytes(self):
+        base = u'test'.decode('ascii')
+        expected = u'test'
+        self._test(base, expected)
+
+    def test_utf8_bytes(self):
+        base = u'test'.decode('utf-8')
```

Comment:
Gahh, python2 and it's enfuriating `str.encode` and `unicode.decode`! those should be `encode`, I've fixed them locally.

The only case we're not hitting (or should be hitting) is the malformed encoding case that causes the LookupError. And I'm not convinced that we should be handling that case really, and if we are going to handle it, we should probably use magic to guess the encoding rather than assuming that it's ascii.